### ISSUE TITLE
Webpack Module Federation Docs

### ISSUE THREAD

**Issue by RyanCCollins** [webpack/webpack.js.org#4283]
## Feature request
My company is evaluating micro-frontend frameworks for a platform. Webpack is high up on the list of potential frameworks, but unfortunately, the webpack module federation docs are not up to the standard of the more stable webpack docs.

**What is the expected behavior?**
- The docs for module federation are missing API Docs containing explanation of the properties on the ModuleFederationPlugin and low-level container plugins.
  - It's not clear how shared modules, exposes and remotes work. The only way to figure it out is to dig into the examples.
  - Not clear what overrides vs. overridables are and how they relate to the plugin.
  - A deeper dive into shared modules, including their optional properties, would be helpful.
  - Other properties should be documented to show their intent and data type.

**What is motivation or use case for adding/changing the behavior?**
Making Webpack module federation adoptable

**How should this be implemented in your opinion?**
Similar to the other webpack API docs 

**Are you willing to work on this yourself?**
no, I don't have the required knowledge or the time unfortunately.  If someone with the knowledge would provide me with details, I can submit a PR.

**Comment by estevaolucas** [comment]
After digging into Module Federation codebase, I realized that https://webpack.js.org/concepts/module-federation/ is presenting outdated information. 

With this PR https://github.com/webpack/webpack/pull/10960, `OverridablesPlugin` got replaced with `SharePlugin`. That means Overriding, OverridablesPlugin, Dynamic Remote Containers sections contain invalid information.

**Comment by jayankmayukh** [comment]
Facing same issue. As of now we are using examples to understand the api, a proper documentation of API is very important.

**Comment by tmkasun** [comment]
Yes same here, there is very little documentation for `ModuleFederationPlugin` itself and its configuration.

**Comment by tetchel** [comment]
for me, I got lost trying to figure out how to use the OverridablesPlugin (https://webpack.js.org/concepts/module-federation/#building-blocks) 
only to find out it was removed in https://github.com/webpack/webpack/pull/10960

**Comment by tetchel** [comment]
I think this is also supposed to be covered under https://github.com/webpack/webpack.js.org/issues/3757
